client/src/GameState.ts

```diff
@@ -99,6 +101,13 @@ export class GameState {
                 this.emitAndPlaceCurrentTetromino();
             }
         });
+
+        this.socket.on("votedTetroToSpawn", (type) => {
+            this.votedTetro = type;
+            setTimeout(() => {
+                this.votedTetro = null;
+            }, 20000);
+        });
```

Comment:
Would it be cleaner for RandomBag to contain the `votedTetroToSpawn` listener and have the `votedTetro` variable? Could also handle the if statement inside of `getNextType()`?

Reply:
I'm not sure if its cleaner but it will make more sense 👍